### PR TITLE
add support for remote_read

### PR DESCRIPTION
**Pull request by lobeck**
required for prometheus 2 migration
https://prometheus.io/docs/prometheus/2.0/migration/#storage

example:

      prometheus::remote_read_configs:
        - url: 'http://somewhere:9090/api/v1/read'

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -144,6 +147,7 @@
   Hash $global_config         = $::prometheus::params::global_config,
   Array $rule_files           = $::prometheus::params::rule_files,
   Array $scrape_configs       = $::prometheus::params::scrape_configs,
+  Array $remote_read_configs  = $::prometheus::params::remote_read_configs,
```
Comment (bastelfreak) [inline manifests/init.pp:150]: Datatypes \o/

File: templates/prometheus.yaml.erb [hunk 2]
```diff
@@ -2,13 +2,15 @@
 <% global_config = scope.lookupvar('::prometheus::global_config') -%>
 <% rule_files = scope.lookupvar('::prometheus::rule_files') -%>
 <% scrape_configs = scope.lookupvar('::prometheus::scrape_configs') -%>
+<% remote_read_configs = scope.lookupvar('::prometheus::remote_read_configs') -%>
```
Comment (bastelfreak) [inline templates/prometheus.yaml.erb:5]: what happens when we get an empty array here?

Comment (lobeck) [reply templates/prometheus.yaml.erb:5]: it ends up in the config as `"remote_read": []` and prometheus just ignores it

**Comment by lobeck** [comment]
the tests seem fine now, but i can't get the debian test to pass, any ideas?

**Comment by bastelfreak** [comment]
Debian is our special snowflake. I've no idea why it never works on the first run, but always on the second.

**Pull request merged**